doc/manual/compiler.rst

```diff
@@ -101,11 +105,19 @@ Kernel code can call host functions without any additional ceremony. However, su
     def return_four() -> TInt32:
         return 4
 
+.. tip::
+    Multiple variables of different types can be sent in one RPC call by returning a tuple, e.g. ::
+
+        def return_many() -> TTuple([TInt32, TFloat, TStr]):
+            return (4, 12.34, "hello",)
+
+    Which can be retrieved from a kernel with ``(a, b, c) = return_many()``
+
 Kernels can freely modify attributes of objects shared with the host. However, by necessity, these modifications are actually applied to local copies of the objects, as the latency of immediate writeback would be unsupportable in a real-time environment. Instead, modifications are written back *when the kernel completes;* notably, this means RPCs called by a kernel itself will only have access to the unmodified host version of the object, as the kernel hasn't finished execution yet. In some cases, accessing data on the host is better handled by calling RPCs specifically to make the desired modifications.
 
 .. warning::
 
-    Kernels *cannot and should not* return lists, arrays, or strings they have created, or any objects containing them; in the absence of a heap, the way these values are allocated means they cannot outlive the kernels they are created in. Trying to do so will normally be discovered by lifetime tracking and result in compilation errors, but in certain cases lifetime tracking will fail to detect a problem and experiments will encounter memory corruption at runtime. For example: ::
+    Kernels *cannot and should not* return lists, arrays, tuples, or strings they have created, or any objects containing them; in the absence of a heap, the way these values are allocated means they cannot outlive the kernels they are created in. Trying to do so will normally be discovered by lifetime tracking and result in compilation errors, but in certain cases lifetime tracking will fail to detect a problem and experiments will encounter memory corruption at runtime. For example: ::
```

Comment:
Are you sure there are memory lifetime issues around tuples? (of course, a tuple containing a type with indirections such as a list won't work any better than a bare list)

Reply:
I think you're right - at least no matter how hard I tried I could not manage to corrupt a tuple. Removed.

I also realize I forgot to mention indexing by a constant .

Reply:
Was the mention in this paragraph meant to be removed as well?

Reply:
It was, sorry about that